From bb9c6941402370821ea73790ec1b9ea00c2a1e84 Mon Sep 17 00:00:00 2001
From: Sreedhar Govardhana <srgovard@amd.com>
Date: Wed, 18 Aug 2021 16:58:21 -0500
Subject: [PATCH] ARM:dts:aspeed: APML over I2C device tree support for
 Onyx/Quartz

APML over I2C is enabled with sbtsi node in dts of Onyx and Quartz.

Signed-off-by: Sreedhar Govardhana <srgovard@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts   |  7 ++++++-
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts | 18 ++++++++++++++----
 2 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 847bb2269567..4f66efc31c1c 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -418,9 +418,14 @@ spd5118_1_5: spd@55 {
 	};
 };

-&i3c4 {
+&i2c2 {
 	// P0 APML
 	status = "okay";
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
 };

 &kcs1 {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index f826b37319e2..b16910083ec6 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -399,14 +399,24 @@ spd5118_3_5: spd@55 {
 	};
 };

-&i3c4 {
-	// P0 APML - need to add mux
+&i2c2 {
+	// P0 APML
 	status = "okay";
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
 };

-&i3c5 {
-	// P1 APML - need to add mux
+&i2c3 {
+	// P1 APML
 	status = "okay";
+
+	sbtsi@48 {
+		compatible = "amd,sbtsi";
+		reg = <0x48>;
+	};
 };

 &gpio0 {
--
2.25.1

